docs/tutorials/rest_api/index.md

```diff
@@ -0,0 +1,412 @@
+> connectors to the same stream table or incremental view.
+
+It will return `HTTP/1.1 201 Created` when successful.
+The pipeline will now be ready to be started.
+
+### Step 4: Starting pipeline
+
+We start the pipeline using:
+
+```
+curl -i -X POST http://localhost:8080/v0/pipelines/sc-pipeline/start
+```
+
+... which will return `HTTP/1.1 202 Accepted` when successful.
+
+Check it is successfully started using:
+
+```
+curl -X GET http://localhost:8080/v0/pipelines/sc-pipeline
+```
+
+... which will have `current_status` set to `Running` when it has started.
+
+> Note: only when a pipeline is started, the connectors are actually used.
+> A pipeline will not start if a connector is unable to connect to its
+> data source or sink (e.g., if a URL is misspelled).
+
+> To restart a pipeline (e.g., to have updates to its program or schema
+> take effect):
+> ```
+> # Shut it down:
+> curl -i -X POST http://localhost:8080/v0/pipelines/sc-pipeline/shutdown
+> # ... wait for the current_status to become Shutdown by checking:
+> curl -X GET http://localhost:8080/v0/pipelines/sc-pipeline
+> # ... and then start:
+> curl -i -X POST http://localhost:8080/v0/pipelines/sc-pipeline/start
+> ```
+
+### Step 5: Pipeline progress
+
+A running pipeline provides a multitude of interesting stats:
+
+```
+curl -X GET http://localhost:8080/v0/pipelines/sc-pipeline/stats
+```
+
+... such as the number of input and processed records.
+The `total_processed_records` should be `9` (3 rows each for part, vendor, and price).
+
+### Step 6: Read data directly from an incremental view
+
+Both input and output connectors are optional, in the sense that input and
+output of data can directly be done using HTTP requests as well.
+
+We can retrieve the incremental view quantiles snapshot using `curl`:
+
+```
+curl -X POST 'http://localhost:8080/v0/pipelines/sc-pipeline/egress/PREFERRED_VENDOR?format=json&mode=snapshot&query=quantiles'
+```
+
+... which for each of the parts will show the preferred vendor.
+
+This includes Gravitech Dynamics (vendor id: 1) for the Warp Core (part id: 2)
+at a price point of 15000.
+
+It is also possible to actively monitor an incremental view for changes
+rather than retrieving a snapshot:
+
+```
+curl -N -X POST 'http://localhost:8080/v0/pipelines/sc-pipeline/egress/PREFERRED_VENDOR?format=json&mode=watch'
+```
+
+Keep this open in a separate terminal for the next step.
+Even if there is no changes it will regularly send an empty message.
+
+### Step 7: Feed data directly into a stream table
+
+It is possible to directly feed data into a stream table. In this case,
+we have HyperDrive Innovations supply the Warp Core at a lower price of 12000:
+
+```
+curl -X 'POST' http://localhost:8080/v0/pipelines/sc-pipeline/ingress/PRICE?format=json \
+-d '{"insert": {"part": 2, "vendor": 2, "price": 12000}}'
+```
+
+Now we can check in the watch terminal that indeed the preferred vendor is changed
```

Comment:
```suggestion
In the other terminal, we can see preferred vendor view output change,
```

Reply:
Changed to:

In the other terminal, we can see the preferred vendor view output change, with a row deletion (with the previous cheapest vendor) and insertion (with the new cheapest vendor).